server/src/test/java/org/elasticsearch/search/profile/query/ProfileCollectorManagerTests.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.profile.query;
+
+import org.apache.lucene.search.Collector;
+import org.apache.lucene.search.CollectorManager;
+import org.apache.lucene.tests.search.DummyTotalHitCountCollector;
+import org.apache.lucene.util.SetOnce;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+public class ProfileCollectorManagerTests extends ESTestCase {
```

Comment:
would it make sense to have a test that simulates profiling a search with and without concurrency? if you call `newIndexSearcher` it will randomize the slices and randomly set the executor

Reply:
Good idea, what should be the goal of that test? Would we want to e.g. prep the searcher with data and then see if all results are there (in this case the top docs?) The profiling times wouldn't make very much sense to check except that they are there.

Reply:
I would test the output of profiling, timing per se does not make sense to check indeed.